llm/rag/batch_compute_embeddings.py

```diff
@@ -0,0 +1,100 @@
+"""
+Use skypilot to launch managed jobs that will run the embedding calculation for RAG.
+
+This script is responsible for splitting the input dataset up among several workers,
+then using skypilot to launch managed jobs for each worker. We use compute_embeddings.yaml
+to define the managed job info.
+"""
+
+#!/usr/bin/env python3
+
+import argparse
+import os
+
+import sky
+
+
+def calculate_job_range(start_idx: int, end_idx: int, job_rank: int,
+                        total_jobs: int) -> tuple[int, int]:
+    """Calculate the range of indices this job should process.
+    
+    Args:
+        start_idx: Global start index
+        end_idx: Global end index
+        job_rank: Current job's rank (0-based)
+        total_jobs: Total number of jobs
+        
+    Returns:
+        Tuple of [job_start_idx, job_end_idx)
+    """
+    total_range = end_idx - start_idx
+    chunk_size = total_range // total_jobs
+    remainder = total_range % total_jobs
+
+    # Distribute remainder across first few jobs
+    job_start = start_idx + (job_rank * chunk_size) + min(job_rank, remainder)
+    if job_rank < remainder:
+        chunk_size += 1
+    job_end = job_start + chunk_size
+
+    return job_start, job_end
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Launch batch RAG embedding computation jobs')
+    parser.add_argument('--start-idx',
+                        type=int,
+                        default=0,
+                        help='Global start index in dataset')
+    parser.add_argument(
+        '--end-idx',
+        type=int,
+        # this is the last index of the reddit post dataset
+        default=109740,
+        help='Global end index in dataset, not inclusive')
+    parser.add_argument('--num-jobs',
+                        type=int,
+                        default=10,
+                        help='Number of jobs to partition the work across')
+    parser.add_argument('--env-path',
+                        type=str,
+                        default='~/.env',
+                        help='Path to the environment file')
+    args = parser.parse_args()
+
+    # Try to get HF_TOKEN from environment first, then ~/.env file
+    hf_token = os.environ.get('HF_TOKEN')
+    if not hf_token:
+        env_path = os.path.expanduser(args.env_path)
+        if os.path.exists(env_path):
+            with open(env_path) as f:
+                for line in f:
+                    if line.startswith('HF_TOKEN='):
+                        hf_token = line.strip().split('=')[1]
+                        break
```

Comment:
nit: we can use `dotenv` to load the env file: https://github.com/skypilot-org/skypilot/blob/73e95ab79bf75923963a93733e54c100d0567e02/sky/client/cli.py#L353-L361